ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/identity_provider_config_service.proto

```diff
@@ -0,0 +1,141 @@
+
+// Identity Provider Config Service makes it possible for participant node administrators
+// to setup and manage additional identity providers at runtime.
+
+// This allows accepting JWTs from issuers which are provided using this API.
+// The user-id provided in the JWT token and `identity_provider_id` will be used to identity the user
+// against the user management database. Users and parties identified by the identity provider
+// created using this API will be isolated from other users and parties having other identity provider.
+// Users and parties without `identity_provider_id` defined are assumed to be using default identity provider,
+// which is provided in the participant node configuration.
+
+// The fields of request messages (and sub-messages) are marked either as ``Optional`` or ``Required``:
+// (1) ``Optional`` denoting the client may leave the field unset when sending a request.
+// (2) ``Required`` denoting the client must set the field to a non-default value when sending a request.
+//
+// An identity provider config resource is described by the ``IdentityProviderConfig`` message,
+// An identity provider config resource, once it has been created, can be modified.
+// In order to update the properties represented by the ``IdentityProviderConfig`` message use the ``UpdateIdentityProviderConfig`` RPC.
+// The only fields that can be modified are those marked as ``Modifiable``.
+service IdentityProviderConfigService {
+
+  // Create a new identity provider configuration.
+  // The request will fail if the maximum allowed number of separate configurations is reached.
+  rpc CreateIdentityProviderConfig (CreateIdentityProviderConfigRequest) returns (CreateIdentityProviderConfigResponse);
+
+  // Get the identity provider configuration data by id.
+  rpc GetIdentityProviderConfig (GetIdentityProviderConfigRequest) returns (GetIdentityProviderConfigResponse);
+
+  // Update selected modifiable attribute of a identity provider config resource described
+  // by the ``IdentityProviderConfig`` message.
+  rpc UpdateIdentityProviderConfig (UpdateIdentityProviderConfigRequest) returns (UpdateIdentityProviderConfigResponse);
+
+  // List all existing identity provider configurations.
+  rpc ListIdentityProviderConfigs (ListIdentityProviderConfigsRequest) returns (ListIdentityProviderConfigsResponse);
+
+  // Delete an existing identity provider configuration.
+  rpc DeleteIdentityProviderConfig (DeleteIdentityProviderConfigRequest) returns (DeleteIdentityProviderConfigResponse);
+
+}
+
+message IdentityProviderConfig {
+  // The identity provider identifier
+  // Must be a valid LedgerString (as describe in ``value.proto``).
+  // Required
+  string identity_provider_id = 1;
+
+  // When set, then the JWT token issued by this identity provider is denied all access to the Ledger API.
+  // Otherwise, the user-id and `identity_provider_id` provided in the JWT token will be used to identity
+  // the user in the user management database.
```

Comment:
```suggestion
  // When set, the callers using JWT tokens issued by this identity provider are denied all access
  // to the Ledger API. Otherwise, the `sub` and `iss` claims provided in the JWT token will be 
  // used to identify the users in the user management database.
```